docs/reference/contrib/settings.md

```diff
@@ -0,0 +1,379 @@
+`settings`, use the provided `{% get_settings %}` template tag.
+
+```html+django
+{% load wagtailsettings_tags %}
+{% get_settings %}
+{{ settings.app_label.GenericSocialMediaSettings.facebook }}
+{{ settings.app_label.SiteSpecificSocialMediaSettings.facebook }}
+```
+
+By default, the tag will create or update a `settings` variable in the
+context. If you want to assign to a different context variable instead, use
+`{% get_settings as other_variable_name %}`:
+
+```html+django
+{% load wagtailsettings_tags %}
+{% get_settings as wagtail_settings %}
+{{ wagtail_settings.app_label.GenericSocialMediaSettings.facebook }}
+{{ wagtail_settings.app_label.SiteSpecificSocialMediaSettings.facebook }}
+```
+
+### Using in Jinja2 templates
+
+Add `wagtail.contrib.settings.jinja2tags.settings` extension to your
+Jinja2 settings:
+
+```python
+TEMPLATES = [
+    ...
+
+    {
+        'BACKEND': 'django.template.backends.jinja2.Jinja2',
+        'APP_DIRS': True,
+        'OPTIONS': {
+            'extensions': [
+                ...
+
+                'wagtail.contrib.settings.jinja2tags.settings',
+            ],
+        },
+    }
+]
+```
+
+Then access the settings through the `settings()` template function:
+
+```html+jinja
+{{ settings("app_label.GenericSocialMediaSettings").facebook }}
+{{ settings("app_label.SiteSpecificSocialMediaSettings").facebook }}
+```
+
+**Note:** Replace `app_label` with the label of the app containing your
+settings model.
+
+If there is no `request` available in the template at all, you can use the
+settings for the default Wagtail site instead:
+
+```html+jinja
+{{ settings("app_label.GenericSocialMediaSettings", use_default_site=True).facebook }}
+{{ settings("app_label.SiteSpecificSocialMediaSettings", use_default_site=True).facebook }}
+```
+
+**Note:** You can not reliably get the correct settings instance for the
+current site from this template tag if the request object is not available.
+This is only relevant for multisite instances of Wagtail.
+
+You can store the settings instance in a variable to save some typing,
+if you have to use multiple values from one model:
+
+```html+jinja
+{% with generic_social_settings=settings("app_label.GenericSocialMediaSettings") %}
+    Follow us on Twitter at @{{ generic_social_settings.facebook }},
+    or Instagram at @{{ generic_social_settings.instagram }}.
+{% endwith %}
+
+{% with site_social_settings=settings("app_label.SiteSpecificSocialMediaSettings") %}
+    Follow us on Twitter at @{{ site_social_settings.facebook }},
+    or Instagram at @{{ site_social_settings.instagram }}.
+{% endwith %}
+```
+
+Or, alternately, using the `set` tag:
+
+```html+jinja
+{% set generic_social_settings=settings("app_label.GenericSocialMediaSettings") %}
+{% set site_social_settings=settings("app_label.SiteSpecificSocialMediaSettings") %}
+```
+
+## Utilising `select_related` to improve efficiency
+
+For models with foreign key relationships to other objects (e.g. pages),
```

Comment:
```suggestion
For models with foreign key relationships to other objects (for example, pages),
```

Reply:
We use `e.g.` elsewhere - what's the reasoning/thinking behind changing this one?

Reply:
TIL! Thanks for the references :)